aiohttp/client_exceptions.py

```diff
@@ -238,17 +239,31 @@ class InvalidURL(ClientError, ValueError):
 
     # Derive from ValueError for backward compatibility
 
-    def __init__(self, url: Any) -> None:
+    def __init__(self, url: Any, description: Optional[str] = None) -> None:
         # The type of url is not yarl.URL because the exception can be raised
         # on URL(url) call
-        super().__init__(url)
+        super_args = [url]
+        if description:
+            super_args.append(description)
+        super().__init__(*super_args)
 
     @property
     def url(self) -> Any:
         return self.args[0]
 
+    @property
+    def description(self) -> Any:
+        return self.args[1]
```

Comment:
This would error if description wasn't passed.
Also return type should not be `Any`.

Reply:
Right, I lost KeyError here - added. There is problem with typing:
`aiohttp/client_exceptions.py:257: error: Returning Any from function declared to return "Optional[str]"`
That's why `def url` has Any return as well. I updated both to match it's real type and added `# type: ignore[no-any-return]` - this seems to be better

Reply:
For url, the argument type is `Any`, so the return type would need to match that. Why it doesn't have a more precise type, I'm not sure, but it's unclear to me whether it should always be `str` or not (I'd need to take a closer look to check).

Reply:
A type ignore comment is fine. Another option would be to cast it, or maybe even annotate the type in the class like `args: Tuple[str, str]`

Reply:
I updated `url` type to `StrOrURL` - `yarl.URL` can be passed to this exception as well. Regarding args - there is no access to overwrite type as its `Exception` thing